Added new doc page for the new date -picker in 12.2

psatyal opened the pull request:
Hi @birthe. This PR adds a whole new section titled "Set and change dates" that documents the new date picker, including the information and warning banners and "Show relations" button.  This will need to be further updated in 12.3 with documentation on non-working days and duration.

I've also updated the scheduling page under the GANTT section with an updated image of the "Manual scheduling" checkbox.

I hope the relative paths and images work fine. Could you please review?

https://community.openproject.org/notifications/details/41752

psatyal committed:
Added new page for date picker, updated GANTT

This commit adds a whole new section titled "Set and change dates" that documents the new date picker, including the information and warning banners and "Show relations" button.

This will need to be further updated in 12.3 with documentation on non-working days and duration.

psatyal committed:
Merge branch 'opf:dev' into dev

psatyal commented:
Trying to link to work package https://community.openproject.org/work_packages/41752/activity

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+| ------------------------------------------------------------ | --------------------------------------------------------- |
+| [Set start and finish dates](#set-start-and-finish-dates)    | How to set start and finish dates, and how to change them |
+| [Scheduling mode](#scheduling-mode)                          | Enabling manual scheduling and what it does               |
+| [Warning and information banners](#warning-and-information-banner) | What the different warning and information banners mean   |
```
This link seems wrong, should be "information-and-warning-banners" just like the headline on this page

as-op replied:
the anchor link is different to the headline, the link should be
 Information and warning banners
`[Warning and information banners](#information-and-warning-banner)`

birthe commented on docs/user-guide/work-packages/README.md:
```diff
@@ -30,12 +30,13 @@ Work packages can be displayed in a projects timeline, e.g. as a milestone or a
-| [Exporting](exporting)                                       | How to export work packages for other tools such as Microsoft Excel? |
-| [Work package relations and hierarchies](work-package-relations-hierarchies) | How to create work package relations and hierarchies?        |
+| [Create a work package](create-work-package)                 | How to create a new work package in OpenProject              |
+| [Set and change dates](set-and-change-dates)                 | How to set and change start and finish dates of work packages in OpenProject |
```
This link seems wrong, should be "set-change-dates" just like the URL (folder name).

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+
+You can open the date picker for a work package from a number of different places:
+
+- By clicking on the date field in the work package [details view](../../work-packages/work-package-views/#work-package-full-screen-view)
```
You are already in /work-packages/... So the link should be (../work-package-views/#work-package-full-screen-view). Is this correct, @as-op ?

as-op replied:
This path is technically working, but just FYI: 
it is navigating two levels up `../../` just to enter `work-packages` again.
Since we are already in sub-folder of `work-packages` It can be shortened to
`[details view](../work-package-views/#work-package-full-screen-view)`

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+You can open the date picker for a work package from a number of different places:
+
+- By clicking on the date field in the work package [details view](../../work-packages/work-package-views/#work-package-full-screen-view)
+- By clicking the the date field in the work pacakge [split screen view](../../work-packages/work-package-views/#work-package-split-screen-view) (from any other view, including [notification center](../../notifications), [team planner](../../team-planner/), [boards](../../agile-boards)...)
```
Same here: (../work-package-views/#work-package-split-screen-view)

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+
+- By clicking on the date field in the work package [details view](../../work-packages/work-package-views/#work-package-full-screen-view)
+- By clicking the the date field in the work pacakge [split screen view](../../work-packages/work-package-views/#work-package-split-screen-view) (from any other view, including [notification center](../../notifications), [team planner](../../team-planner/), [boards](../../agile-boards)...)
+- By clicking the start or finish date fields in [list view](../../work-packages/work-package-views/#work-package-list-view)
```
Same here: (../work-package-views/#work-package-list-view)

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+
+![clear-date-field](clear-date-field.png)
+
+Date changes are documented in the work package [Activity](../../../getting-started/work-packages-introduction/#activity-of-work-packages).
```
Maybe link to: ../../activity/ ?

psatyal replied:
Indeed, this is better!

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+
+![manual-scheduling-checkbox](manual-scheduling-checkbox.png)
+
+Checking  "Manual scheduling" enables [manual scheduling mode](../../gantt-chart/scheduling#manual-scheduling-mode) for the work package. This ignores work package relations such that you're able to select any start and end dates, even ones that would normally be restricted due to follows/proceeds relationships. 
```
I think it has to be ../../gantt-chart/scheduling/#manual-scheduling-mode - with a "/" before the #. Is this correct, @as-op ?

as-op replied:
It's better for clarity and SEO, but it's not technically necessary.

psatyal replied:
Will update nevertheless :)

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+
+## Information and warning banners
+
+OpenProject 12.2 introduced helpful banners to give you important contextual information when changing dates on work packages with [relations](../../work-packages/work-package-relations-hierarchies).
```
You are already in work-packages. (../work-package-relations-hierarchies/)

psatyal replied:
Updated.

birthe commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+The information and warning banners also feature a **Show Relations** button. Clicking on this will open a new tab that displays work packages with direct relations to the current work package in [Gantt view](../../gantt-chart), in hierarchy mode. 
+
+> **Info**: This preview is intended to give a quick overview of only _direct_ relations that might affect scheduling options for the current work package. It does not show second-level relations (and above). To get a full overview, please use the project work package [list view](../../work-packages/work-package-views/#work-package-list-view) or [Gantt view](../../gantt-chart) with your desired [ ...
+
```
../work-package-table-configuration/

as-op commented on docs/user-guide/work-packages/README.md:
```diff
@@ -30,12 +30,13 @@ Work packages can be displayed in a projects timeline, e.g. as a milestone or a
-| [Exporting](exporting)                                       | How to export work packages for other tools such as Microsoft Excel? |
-| [Work package relations and hierarchies](work-package-relations-hierarchies) | How to create work package relations and hierarchies?        |
+| [Create a work package](create-work-package)                 | How to create a new work package in OpenProject              |
+| [Set and change dates](set-and-change-dates)                 | How to set and change start and finish dates of work packages in OpenProject |
```
The path fragment is different, the link should be 
`[Set and change dates](set-change-dates)`

psatyal replied:
Fixed, thank you.

as-op commented on docs/user-guide/work-packages/set-change-dates/README.md:
```diff
@@ -0,0 +1,138 @@
+
+To clear a date field, simply click on the **clear icon** ("×") icon when the field is in focus.
+
+![clear-date-field](clear-date-field.png)
```
Just an FYI: please use full english text in the [image descriptions](https://www.openproject.org/docs/development/contribution-documentation/documentation-style-guide/#add-the-image-link-to-content), since they are read aloud by screenreaders, the dashes don't do good in most applications.

psatyal replied:
Ok, added full descriptions. I was going using existing documentation as a guide, so some existing pages need to be updated too it looks like.

birthe reviewed (commented):
Can you please have a look at the links @as-op if that's correct now?

birthe commented:
Hi @psatyal sorry, I didn't submit the review correctly, so Andrej added the same comments before I confirmed my review :-) So you now have it twice ;-)

psatyal committed:
Updated the text based based on review

Fixed relative paths, improved image alt text and made a few minor changes to improve the flow of the text.

psatyal committed:
Fixed link to Set and change dates page

Fixed link

psatyal committed:
Added text description to the new image

We should probably update the other images too

psatyal reviewed (commented):
Hi @birthe and @as-op, thank you for your review and comments. I've made the necessary changes. I did it manually by editing the file instead of accepting changes because I had a few other things to fix too. Could you review again please?

as-op commented:
Recreated & merged into release/12.2 at https://github.com/opf/openproject/pull/11061
